### PR TITLE
#165890222 following and unfollowing users

### PR DESCRIPTION
**Pull request by joelmugaya**
#### What does the PR do?
- Allows the user to follow, unfollow and list followers and users he follows.
#### Description of Task to be completed?
- Given a user is logged in, he can follow another user
- He can view users he follows or that another user follows
- He can view users that follow him or that follow another user
#### Tasks to be completed
- update profile model to include following functionality
- update profile serializer to add a following field to the profile serialized data
- write tests for the following functionality
- add endpoints for the follow functionality to the url.py file
- write functions to handle the follow functionality
#### How should this be manually tested?
- to follow a user `POST /api/profiles/<username>/follow/` returns `"You are following username."`
- on getting the profile `GET /api/users/profile/<username>/` returns `following: True`
- to unfollow a user `DELETE /api/profiles/<username>/follow/`
- to get users followers `GET /api/profiles/<username>/followers/`
- to get users following a user `GET /api/profiles/<username>/following/`

[finishes #165890222]

### CODE REVIEW

**Review by engjames — APPROVED** [review]
LGTM

**Review by 3Nakajugo — APPROVED** [review]
tested the pull request on Heroku and it works

**Review by mwinel — APPROVED** [review]
LGTM.

File: authors/apps/profiles/views.py [hunk 1]
```diff
@@ -41,13 +43,86 @@ def put(self, request, *args, **kwargs):
...
+    def get(self, request, username):
+        # get the user's followers
+        user = get_user_object(username)
+        followers = Follow.objects.filter(followed=user)
+        profiles = []
+        for value in followers:
+            profile = UserProfile.objects.get(user=value.follower)
+            profile = FetchUserProfileSerializer(profile)
+            profiles.append(profile.data)
+        return Response(profiles, status=status.HTTP_200_OK,)
```
Comment (archibishop) [inline authors/apps/profiles/views.py:113]: @joelmugaya We should return a message `You have no followers` when the list being returned empty.

File: authors/apps/profiles/views.py [hunk 2]
```diff
@@ -41,13 +43,86 @@ def put(self, request, *args, **kwargs):
...
+    def get(self, request, username):
+        # get users this user follows
+        user = get_user_object(username)
+        followings = Follow.objects.filter(follower=user)
+        profiles = []
+        for value in followings:
+            profile = UserProfile.objects.get(user=value.followed)
+            profile = FetchUserProfileSerializer(profile)
+            profiles.append(profile.data)
+        return Response(profiles, status=status.HTTP_200_OK,)
```
Comment (archibishop) [inline authors/apps/profiles/views.py:130]: @joelmugaya We should return a message `You are following anyone.` when the list being returned empty.

**Pull request merged**